editor/src/components/canvas/canvas-strategies/flow-reorder-helpers.ts

```diff
@@ -33,6 +29,66 @@ export function isValidFlowReorderTarget(
   }
 }
 
+export function areAllSiblingsInOneDimension(
+  target: ElementPath,
+  metadata: ElementInstanceMetadataMap,
+): boolean {
+  const targetElement = MetadataUtils.findElementByElementPath(metadata, target)
+  const siblings = MetadataUtils.getSiblings(metadata, target) // including target
+  if (targetElement == null || siblings.length === 1) {
+    return false
+  }
+
+  const targetDirection = getElementDirection(targetElement)
+
+  let allHorizontalOrVertical = true
+  let frames: Array<CanvasRectangle> = []
+  fastForEach(siblings, (sibling) => {
+    if (isValidFlowReorderTarget(sibling.elementPath, metadata)) {
+      if (getElementDirection(sibling) !== targetDirection) {
+        allHorizontalOrVertical = false
+      }
+      if (sibling.globalFrame != null) {
+        frames.push(sibling.globalFrame)
+      }
+    }
+  })
+
+  return allHorizontalOrVertical && areNonWrappingSiblings(frames, targetDirection)
+}
+
+function areNonWrappingSiblings(
+  frames: Array<CanvasRectangle>,
+  layoutDirection: 'horizontal' | 'vertical',
+): boolean {
+  if (layoutDirection === 'horizontal') {
+    // all elements are on the right side of the previous sibling
+    return frames.every((frame, i) => {
+      if (i === 0) {
+        return true
+      } else {
+        const prevFrame: CanvasRectangle = frames[i - 1]
+        return frame.x > prevFrame.x
```

Comment:
if you put the if (layoutDirection === 'horizontal') inside here that saves you from having to write the loop code twice

Reply:
fixed